### PR TITLE
[WX]add maxlines to Text

### PR DESCRIPTION
**Pull request by wx-uf**
**Changes**
Description of changes...

- [keep | delete] branch

**Changelog entry**

- add new props maxLines to Text in order to specify the number of lines visible, the additional lines will be truncated

**Additional information**

- You may refer to this [ticket](https://jira.ship.gov.sg/browse/CCUBE-382)
- Any other information you would like to include

### CODE REVIEW

File: src/text/helper.ts [hunk 1]
```diff
@@ -77,6 +77,18 @@ const getFontWeight = (weight: TextWeight) => {
 // =============================================================================
 // TEXT STYLE
 // =============================================================================
+const getMaxLinesLineStyle = (maxLines: number | undefined = undefined) => {
```
Comment (qroll) [inline src/text/helper.ts:80]: the internal function can be more explicit about the number of args required

```suggestion
const getMaxLinesLineStyle = (maxLines: number | undefined) => {
```

File: src/text/helper.ts [hunk 2]
```diff
@@ -77,6 +77,18 @@ const getFontWeight = (weight: TextWeight) => {
...
+    if (maxLines > 0) {
+        return css`
+            display: -webkit-box !important;
```
Comment (qroll) [inline src/text/helper.ts:83]: is this `!important` required?

File: src/text/helper.ts [hunk 3]
```diff
@@ -77,6 +77,18 @@ const getFontWeight = (weight: TextWeight) => {
...
+            -webkit-box-orient: vertical;
+            -webkit-line-clamp: ${maxLines};
+            line-clamp: ${maxLines};
```
Comment (qroll) [inline src/text/helper.ts:86]: I think `line-clamp` is not an official css property yet, is this more for future compatibility?

File: src/text/helper.ts (hunk 2 above)
Comment (wx-uf) [reply src/text/helper.ts:83]: yes, need, otherwise display: 'block' overwrite it based on the testing.

File: src/text/helper.ts (hunk 3 above)
Comment (wx-uf) [reply src/text/helper.ts:86]: yes, pls advice remain or remove

Comment (qroll) [reply src/text/helper.ts:86]: I noticed the styled components plugin recognises this property so it's fine

File: src/text/helper.ts (hunk 2 above)
Comment (wx-uf) [reply src/text/helper.ts]: removed and code submitted

**Review by qroll — APPROVED** [review]

**Pull request merged**